test/realtime/message.test.js

```diff
@@ -1183,5 +1183,136 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
           });
       });
     }
+
+    it('subscribes to filtered channel', function (done) {
+      var testData = [
+        {
+          name: 'filtered',
+          data: 'These headers match the JMESPath expression so this message should not be filtered',
+          extras: {
+            headers: {
+              name: 'Lorem Choo',
+              number: 26095,
+              bool: true,
+            },
+          },
+        },
+        {
+          name: 'filtered',
+          data: 'Random data with no extras for filter',
+        },
+        {
+          name: 'filtered',
+          data: 'This message should also not be filtered',
+          extras: {
+            headers: {
+              name: 'John Bull',
+              number: 26095,
+              bool: false,
+            },
+          },
+        },
+        {
+          name: 'filtered',
+          data: 'No header on message',
+        },
+        {
+          name: 'filtered',
+          data: 'Cannot be filtered because it does not meet filter condition on headers.number',
+          extras: {
+            headers: {
+              name: 'John Bull',
+              number: 12345,
+              bool: false,
+            },
+          },
+        },
+        {
+          name: 'end',
+          data: 'last message check',
+        },
+      ];
+
+      var filterOption = {
+        filter: 'name == `"filtered"` && headers.number == `26095`',
+      };
+
+      try {
+        /* set up realtime */
+        var realtime = helper.AblyRealtime({ key: helper.getTestApp().keys[5].keyStr });
+        var rest = helper.AblyRest();
+
+        realtime.connection.on('connected', function () {
+          var rtFilteredChannel = realtime.channels.getDerived('chan', filterOption);
+          var rtUnfilteredChannel = realtime.channels.get('chan');
+
+          var filteredMessages = [];
+          var unFilteredMessages = [];
+          /* subscribe to event */
+          rtFilteredChannel.subscribe(function (msg) {
+            try {
+              // Push received filtered messages into an array
+              filteredMessages.push(msg);
+            } catch (err) {
+              closeAndFinish(done, realtime, err);
+              return;
+            }
+          });
+
+          rtUnfilteredChannel.subscribe(function (msg) {
+            try {
+              // Push received unfiltered messages into an array
+              unFilteredMessages.push(msg);
+            } catch (err) {
+              closeAndFinish(done, realtime, err);
+              return;
+            }
+          });
+
+          // Subscription to check all messages were received as expected
+          rtUnfilteredChannel.subscribe('end', function (msg) {
+            try {
+              expect(msg.data).to.equal(testData[5].data, 'Unexpected msg data received');
```

Comment:
minor, but probably more robust to do `testData[testData.length-1]` here in case we add new messages